kernel/src/osquery.cpp

```diff
@@ -62,6 +66,63 @@ static struct osquery {
   .major_number = OSQUERY_MAJOR
 };
 
+static int fileop_scope_callback(kauth_cred_t credential,
+                                 void *idata,
+                                 kauth_action_t action,
+                                 uintptr_t arg0,
+                                 uintptr_t arg1,
+                                 uintptr_t arg2,
+                                 uintptr_t arg3) {
+  vnode_t vp = (vnode_t)arg0;
+  if (action == KAUTH_FILEOP_EXEC && vp != NULL) {
+    // Someone is executing a file.
+    int path_len = MAXPATHLEN;
+
+    osquery_process_event_t *e =
+      (osquery_process_event_t *)osquery_cqueue_reserve(&osquery.cqueue,
+                                                        OSQUERY_PROCESS_EVENT);
+    if (e) {
+      e->owner_uid = 0;
+      e->owner_gid = 0;
+      e->mode = -1;
+      vfs_context_t context = vfs_context_create(NULL);
+      if (context) {
+        struct vnode_attr vattr = {0};
+        VATTR_INIT(&vattr);
+        VATTR_WANTED(&vattr, va_uid);
+        VATTR_WANTED(&vattr, va_gid);
+        VATTR_WANTED(&vattr, va_mode);
+
+        if (vnode_getattr(vp, &vattr, context) == 0) {
+          e->owner_uid = vattr.va_uid;
+          e->owner_gid = vattr.va_gid;
+          e->mode = vattr.va_mode;
+        }
+
+        vfs_context_rele(context);
+      }
+
+      e->uid = kauth_cred_getruid(credential);
+      e->euid = kauth_cred_getuid(credential);
+
+      e->gid = kauth_cred_getrgid(credential);
+      e->egid = kauth_cred_getgid(credential);
+
+      vn_getpath(vp, e->path, &path_len);
```

Comment:
If you supply a reference to a `path_len = 1024` will this only fill in 1024 bytes of a path? I'm guessing if it did, it would be the least significant 1024 bytes. 

I ask because we could limit the need for a `4k` name down to something more likely, yet less robust.